services/httpd/handler.go

```diff
@@ -1812,6 +1828,16 @@ func convertToEpoch(r *query.Result, epoch string) {
 	}
 }
 
+func convertToRfc3339Nano(r *query.Result) {
+	for _, s := range r.Series {
+		for _, v := range s.Values {
+			if ts, ok := v[0].(time.Time); ok {
+				v[0] = ts.Format(time.RFC3339Nano)
+			}
+		}
+	}
+}
+
```

Comment:
Not required, but the time format (`time.RFC3339Nano`) could be made a parameter instead of hard-coded. The call would then be something like `convertToTimeFormat(r, time.RFC3339Nano)`. The code would be ready to add different time formats later if requested.

`convertToEpoch` is sufficiently different that trying to combine a `convertToTimeFormat` with it would probably make the code harder to read. It's possible by passing the conversion function as a function parameter, but I don't think that would really improve the code.